### PR TITLE
Split Nodes to separate files and add more TypeScript

### PR DESCRIPTION
**Pull request by LucasMatuszewski**
Hey

It's a very simple maintenance commit (but a lot of changes).
I should ask if you want this, but hopefully, you will like what I did ;)

While I was reading the code and trying to understand how it works I decided to add more TypeScript and split nodes to separate files, because:

1. nodes' code will grow and we will have to split it sooner or later,
2. without TS it's risky and we can see more bugs, and we should start thinking about data types asap,
3. the above changes will become more complicated with time, so I suggest doing this now.

I used Github Copilot Workspace (just got access recently, so wanted to try it out ;)
Of course, I checked all the code manually and made a build. All works.

But please, double-check if I/Copilot didn't delete anything.
For example, I had to manually revert `languages` (Copilot didn't notice them and I had an undefined variable)

For your consideration:
- should we delete `TextToSpeechNode`? (now we have `TtsNode`, but both were in the nodes.js file)
- I have added TODOs in TextNode - sorry, I didn't fully understand why you created this `customWidget` and how this component works, so I didn't manage to type it.
- I would add pretierrc file or some other formatting standardization file to prevent autoformatting changes (sorry, I made a mess in a few places with my Prettier config... if you don't want that, I can try to manually revert some of them)

My next PR will cover some functionality, this was a warmup ;)

Lucas

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -28,11 +28,48 @@ export type exportWorkflowType = (graph: LGraph) => Workflow;
 export type loadWorkflowType = (data: Workflow) => void;
 
 export enum Language {
```
Comment (LucasMatuszewski) [inline src/types.ts:30]: Sorry, wrong name of the commit ;)

**Comment by LucasMatuszewski** [comment]
Up :)
Please let me know if you want me to change something in this PR and, generally, if my changes align with your plans.

**Comment by vipyne** [comment]
thank you @LucasMatuszewski !  this is ambitious and awesome.  I'll let other maintainers weigh in on the TS part (cc @hyypeman) but personally, I'm a little weary of it.  Happy to hear more arguments for it though, especially if you have a lightweight dev flow you like.  If we do want to go that route, at a minimum we should update the README with transpilation instructions, etc. 

In general, I am all for some housecleaning before we get much further.  I like splitting the nodes into separate files.  Also your mention of setting up prettier linting.

I propose the following:

1. a PR that introduces prettier and lints all code as is.
2. a PR that naively splits the current node file into multiple files without changing anything (ie, hold off on TS).
by then, we should have a decision if TS is something we want to implement at this time.  the nodes definitely need to be updated to reflect all of pipecat's options, so we should update them regardless once they are in separate files.

thoughts?

**Comment by hyypeman** [comment]
Ah I wish I had seen this before. I might have just broken the abstractions. Love what you did Lucas - agree with Vanessa here. I have no real opinion on ts or not. I'm good with either. But I did update the abstractions (a lot of what you suggested should be updated with the new changes) so if you don't mind taking a look at the new nodes -- much simpler, that would be greatly appreciated!

**Comment by LucasMatuszewski** [comment]
Hey, sorry for the delay!
I totally understand, I didn't ask first, so it was at my risk.
Just did it as a way to learn your code and to test the GitHub Copilot Workspace ;)

So I close this PR and will see where I can help more (but in a few weeks, I need to hold with AI integrations and focus on marketing and Landing Page to make money first ;)

**Comment by LucasMatuszewski** [comment]
> Ah I wish I had seen this before. I might have just broken the abstractions. Love what you did Lucas - agree with Vanessa here. I have no real opinion on ts or not. I'm good with either. But I did update the abstractions (a lot of what you suggested should be updated with the new changes) so if you don't mind taking a look at the new nodes -- much simpler, that would be greatly appreciated!

Ahh, I misunderstood you @hyypeman  !
I've analyzed your last PRs and I see you refactored nodes, but didn't split them into separate files. Ok, I'm on it :)

>  I'm a little weary of it. Happy to hear more arguments for it though

@vipyne To be honest, for me, TS is just a must-have, it helps a lot when the application grows, especially with the data-intensive app (like this one) where we have a lot of forms and users are providing data, and you get a lot of data from APIs. It will save you a lot of trouble with debugging in the long run and should be done ASAP, to have benefits faster and don't have to refactor a lot of code later on.

> especially if you have a lightweight dev flow you like. If we do want to go that route, at a minimum we should update the README with transpilation instructions, etc.

I know only the basics of Vue (React and Next are my tools), but AFAIK it handles TS for us without any additional work on our side. You just build and it will transpile and produce JS bundles. Actually, we already have some files with TS inside them (also Vue files, like in `App.vue` where you have `<script lang="ts">` = TypeScript), and Vue is already configured for TS. I did a production build on my PR branch and it all just worked without any issues.

But ok, I got it. I will just split the files first, in separate PR.
Then let me know if I can add TS in a separate PR.

**Pull request closed without merging**